### ISSUE TITLE
calculate anomaly score waste lots of GPU memory?

### ISSUE THREAD

**Issue by letmejoin** [AlessioGalluccio/FastFlow#9]
z_grouped = torch.cat(test_z, dim=0).view(-1, c.n_transforms_test, c.n_feat)
        anomaly_score = t2np(torch.mean(z_grouped ** 2, dim=(-2, -1)))
train.py line130, 131, when calculate the anomaly_score, much GPU memory needed. And this is can be solved by calculating along with every loop at line 109-112, and then merage the results when the loop is over.

**Comment by AlessioGalluccio** [comment]
Hi @letmejoin, thank you for the suggestion! I'll try to implement it as soon as possible.
Best,
Alessio

**Comment by balablb** [comment]
> z_grouped = torch.cat(test_z, dim=0).view(-1, c.n_transforms_test, c.n_feat) anomaly_score = t2np(torch.mean(z_grouped ** 2, dim=(-2, -1))) train.py line130, 131, when calculate the anomaly_score, much GPU memory needed. And this is can be solved by calculating along with every loop at line 109-112, and then merage the results when the loop is over.

Can you share the modified code，please~

**Comment by AlessioGalluccio** [comment]
It would be very useful if you could share your code, @letmejoin

**Comment by AlessioGalluccio** [comment]
I updated the code. Let me know if there are problems now @letmejoin @balablb

**Comment by letmejoin** [comment]
> I updated the code. Let me know if there are problems now  @balablb

oh, I will test your code as soon. Thks.

**Comment by balablb** [comment]
> I updated the code. Let me know if there are problems now @letmejoin @balablb 

it works ~

**Comment by AlessioGalluccio** [comment]
Thank you very much, I'll close the issue

**Comment by gaowq2017** [comment]
Hi, The code change of calculating anormaly_score is error, and the true code can reffer to [CSFlow ](https://github.com/marco-rudolph/cs-flow/blob/main/train.py)
